### PR TITLE
feat(Designer): Adding managed connector support for hybrid LA

### PR DESCRIPTION
**Pull request by mcaupybugs**
## Requirement Checklist

* [ ] The commit message follows our guidelines
* [ ] Tests for the changes have been added (for bug fixes or features)
* [ ] Docs have been added or updated (for bug fixes or features)

## Type of Change

* [ ] Bug fix
* [x] Feature
* [ ] Other

## Impact of Change

* [ ] **This is a breaking change.**

### CODE REVIEW

File: apps/Standalone/src/designer/app/AzureLogicAppsDesigner/Queries/FetchHybridApps.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const useFetchHybridApps = () => {
       if (!environment.armToken) {
         return [];
       }
-      const query = `resources | where type =~ "microsoft.app/containerApps" and kind contains "workflowapp"`;
```
Comment (Eric-B-Wu) [inline apps/Standalone/src/designer/app/AzureLogicAppsDesigner/Queries/FetchHybridApps.ts:13]: when pushing to main, we should remove this line; we'll operate under the assumption that the ARG change will be merged soon

File: libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts [hunk 2]
```diff
@@ -340,7 +341,21 @@ export class StandardConnectionService extends BaseConnectionService implements
...
+        throw new Error(
+          intl.formatMessage({
+            defaultMessage: 'App identity is not configured on the logic app environment variables.',
+            id: 'WnU9v0',
+            description: 'Error message when no app identity is added in environment variables',
+          })
+        );
+      }
+      identityDetailsForApiHubAuth = { principalId: identity.principalId, tenantId: identity.tenantId };
+    }else{
```
Comment (Eric-B-Wu) [inline libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts:356]: nit: seems to be some formatting issues on this PR

File: libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts [hunk 3]
```diff
@@ -321,14 +321,15 @@ export class StandardConnectionService extends BaseConnectionService implements
...
+      baseUrl
     } = this._options;
     if (!isArmResourceId(connection.id) || !workflowAppDetails) {
       return;
     }
 
     const intl = getIntl();
 
-    if (!isIdentityAssociatedWithLogicApp(workflowAppDetails.identity)) {
+    if (!isHybridLogicApp(baseUrl) && !isIdentityAssociatedWithLogicApp(workflowAppDetails.identity)) {
```
Comment (Eric-B-Wu) [inline libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts:332]: can we create a separate helper function to validate identity validity?

something like 
```
private validateLogicAppIdentity(baseUrl: string, identity: Identity | undefined) {
  if (!isHybridLogicApp(baseUrl) && !isIdentityAssociatedWithLogicApp(identity)) {
    throw new Error(
      intl.formatMessage({
        defaultMessage: 'A managed identity is not configured on the logic app.',
        id: 'WnU9v0',
        description: 'Error message when no identity is associated',
      })
    );
  }

  if (isHybridLogicApp(baseUrl)) {
    if (!identity?.principalId || !identity?.tenantId) {
      throw new Error(
        intl.formatMessage({
          defaultMessage: 'App identity is not configured on the logic app environment variables.',
          id: 'WnU9v0',
          description: 'Error message when no app identity is added in environment variables',
        })
      );
    }
  }
}
```

**Review by Eric-B-Wu — CHANGES_REQUESTED** [review]

File: libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts [hunk 4]
```diff
@@ -410,6 +410,30 @@ export class StandardConnectionService extends BaseConnectionService implements
     });
   }
 
+  private validateLogicAppIdentity(baseUrl: string, identity: Identity | undefined) {
+    if (!isHybridLogicApp(baseUrl) && !isIdentityAssociatedWithLogicApp(identity)) {
```
Comment (Eric-B-Wu) [inline libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts:414]: nit let's reverse this so we don't check isHybridLogicApp twice:

```
private validateLogicAppIdentity(baseUrl: string, identity: Identity | undefined) {

    if (isHybridLogicApp(baseUrl) && (!identity?.principalId || !identity?.tenantId)) {
      throw new Error(
        intl.formatMessage({
          defaultMessage: 'App identity is not configured on the logic app environment variables.',
          id: 'WnU9v0',
          description: 'Error message when no app identity is added in environment variables',
        })
      );
  }
  if (!isIdentityAssociatedWithLogicApp(identity)) {
    throw new Error(
      intl.formatMessage({
        defaultMessage: 'A managed identity is not configured on the logic app.',
        id: 'WnU9v0',
        description: 'Error message when no identity is associated',
      })
    );
  }
}
```

Comment (mcaupybugs) [reply libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts]: done

File: libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts (hunk 2 above)
Comment (mcaupybugs) [reply libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts]: Addressed it

File: apps/Standalone/src/designer/app/AzureLogicAppsDesigner/Queries/FetchHybridApps.ts (hunk 1 above)
Comment (mcaupybugs) [reply apps/Standalone/src/designer/app/AzureLogicAppsDesigner/Queries/FetchHybridApps.ts]: reverted

**Pull request merged**